From dfeecd0e77446e55e6d5e5ac5a58089ae84236a3 Mon Sep 17 00:00:00 2001
From: Ramanathan Choodamani <rchoodam@qti.qualcomm.com>
Date: Thu, 24 Apr 2025 14:39:55 -0700
Subject: [PATCH] CP:wifi:ath12k:Fix link_id exceeding issue if link_id is 15

If link_id is 15 during scan, mac_vdev_create complains
link_id to be exceeding or being equal to the max range.
Change the logic to be returning error only if link_id exceeds
the range.

Signed-off-by: Ramanathan Choodamani <rchoodam@qti.qualcomm.com>
---
 drivers/net/wireless/ath/ath12k/mac.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/net/wireless/ath/ath12k/mac.c b/drivers/net/wireless/ath/ath12k/mac.c
index 3ee655d..9563d27 100644
--- a/drivers/net/wireless/ath/ath12k/mac.c
+++ b/drivers/net/wireless/ath/ath12k/mac.c
@@ -8881,7 +8881,7 @@ int ath12k_mac_vdev_create(struct ath12k *ar, struct ath12k_link_vif *arvif)
 	else
 		link_id = arvif->link_id;
 
-	if (link_id >= ARRAY_SIZE(vif->link_conf)) {
+	if (link_id > ARRAY_SIZE(vif->link_conf)) {
 		ath12k_warn(ar->ab, "link_id %u exceeds max valid links for vif %pM\n",
 			    link_id, vif->addr);
 		return -EINVAL;
-- 
2.34.1

